apps/protocol-frontend/src/hooks/useContributionBulkMint.ts

```diff
@@ -0,0 +1,114 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import useGovrnToast from '../components/toast';
+import { useNetwork, useSigner } from 'wagmi';
+import { useUser } from '../contexts/UserContext';
+import { networks } from '../utils/networks';
+import { ethers } from 'ethers';
+import { ChainIdError } from '@govrn/protocol-client';
+import pluralize from 'pluralize';
+
+type BulkMintOptions = {
+  ipfsContentUri: string;
+  engagementDate: Date | string;
+  attestations: { id: number }[];
+  guildName: string;
+  name: string;
+  action: string;
+  details?: string | null;
+  id: number;
+  proof?: string | null;
+  txHash?: string | null;
+  user: { id: number };
+  date_of_submission: Date | string;
+  activityTypeId: number;
+  status: { id: number; name: string };
+};
+
+const useContributionBulkMint = () => {
+  const toast = useGovrnToast();
+  const queryClient = useQueryClient();
+  const { data: signer } = useSigner();
+  const { chain } = useNetwork();
+  const { userData, govrnProtocol: govrn } = useUser();
+
+  const { mutateAsync, isLoading, isError, isSuccess } = useMutation(
+    ['bulk-mint-contribution'],
+    async (contributions: BulkMintOptions[]) => {
+      if (!(signer && chain?.id && userData)) {
+        throw new Error('What should the message be!?');
+      }
+
+      return await govrn.contribution.bulkMint(
+        {
+          address: networks[chain?.id].govrnContract,
+          chainId: chain?.id,
+          name: networks[chain?.id].name,
+        },
+        signer,
+        userData.address,
+        contributions.map(c => ({
+          id: c.id,
+          activityTypeId: c.activityTypeId,
+          userId: userData.id,
+          args: {
+            detailsUri: ethers.utils.toUtf8Bytes(c.ipfsContentUri),
+            dateOfSubmission: new Date(c.date_of_submission).getTime(),
+            dateOfEngagement: new Date(c.engagementDate).getTime(),
+          },
+          name: ethers.utils.toUtf8Bytes(c.name),
+          details: ethers.utils.toUtf8Bytes(c.details || ''),
+          proof: ethers.utils.toUtf8Bytes(c.proof || ''),
+        })),
+      );
+    },
+    {
+      onSuccess: result => {
+        /* Failed writes/updates to db are included in result. */
+
+        const minted = result.filter(i => i.status === 'fulfilled');
+
+        if (minted.length > 0) {
+          queryClient.invalidateQueries(['contributionList']);
+          queryClient.invalidateQueries(['contributionInfiniteList']);
```

Comment:
Should we be awaiting these

Reply:
- I think this is kinda of fire-and-forget promise. We don't care about the result as much as triggering fetching it. 
- It would propagate errors of these queries here, I made example [here](https://www.typescriptlang.org/play?#code/MYewdgzgLgBADgJxAWwJYQKYAYCMMC8MAhhAJ5jAwAUAlAQHwwDeMAUDDKJCADYYB0PEAHMqAIgBUAOQDKATSkBhGAFoYAFQAW6GAEkYABSRpMMOSACuYmgG52MBBigWEYQ8fQDHEXgDcM4tLySqoa2hB67iieZpbWdgC+rKxc0PAemFgATATEZBTUdPiMLPapvAIYCEgI4lIgUSYYEQBGpDAymkTkGDAA6kQIACbx9lCaSADuMGAY0wCi1SC1YvWNnq3tnd2z-YMjtqxJKeBpAGYWPDxnqFcAsr2EJOSUtAzMbBxEk0SosIjRTI4Wh2L4-P7pQHYLIg+yOZyuGBiPpdKAYfwIACEYiOyQuVxu9wCNH44wwYCovne5T4ghElJoNFY-GARCgwE0VEejBpAiEojEAHkAKrqGS6AAi8wAXGIADQwDCM5KsIA)

Reply:
Cool, yea agreed